modules/gmake2/gmake2_cpp.lua

```diff
@@ -162,7 +162,19 @@
 		end
 
 		-- process custom build commands.
-		if fileconfig.hasCustomBuildRule(filecfg) then
+		if filecfg.buildaction == "Copy" then
```

Comment:
Would it be desirable to support having both?

Reply:
Yeah, what I'm asking is would it be desirable to be an if instead of else if so a script could provide both a build action and a custom build rule and both are invoked?

Reply:
Unsure what other generators do in that case.

Reply:
Yeah, just wondering if this may break existing code that uses "Copy" (but does nothing if gmake2 is used) and has a custom build rule. If there are scripts like that, "Copy" will be used and the custom build rule will be dropped.

Reply:
I'd expect that to be the case, as well. Is there any downside to allowing both custom rules and copy build actions? If not, I think I'd prefer both are there/copy doesn't "overwrite" the custom build rule.

Reply:
Ok, I will setup a project which does both, and check current behavior of other generators.
and report back.

Reply:
I have checked:
- vs*/ninja only does the custom build part.
- codelite does only the copy part (add recently by #2171)
- Others doesn't support `buildaction "Copy"`.

`buildaction` comes from vs, so I propose to mimic vs behavior (also for codelite, so change order of the if/elseif).
Note that directly from vs IDE, we cannot have both at the same time.

Reply:
updated to have same behavior than vs.
test project
https://github.com/Jarod42/premake-sample-projects/tree/customcommand-with-buildaction-copy/projects/project-customcommand-with-buildaction-copy added.